docs/java-run-your-first-app.md

```diff
@@ -0,0 +1,160 @@
+
+Click the RunId. Here we can view everything we want to know about the state of the execution of the code we told the server to run, such as what parameter values it was given, timeout configurations, scheduled retries, number of attempts, stack traceable errors, and more.
+
+![Money transfer empty poller list](/img/docs/web-ui-money-transfer-workflow-details.png)
+
+So, the Workflow is running, but why hasn't the Workflow and Activity code executed yet? Click on the Task Queue name to view active "Pollers" registered to handle these Tasks. The list will be empty. There are no Workers polling the Task Queue!
+
+![Money transfer empty poller list](/img/docs/web-ui-empty-pollers.png)
+
+### The Worker
+
+As mentioned earlier, a Worker is responsible for actually executing pieces of Workflow code. It listens to a Task Queue, polling for Tasks. A Task identifies a function/method, which the Worker executes, returning any result back to the Temporal server. This is what our Worker looks like:
+
+<!--SNIPSTART money-transfer-project-template-java-worker-->
+<!--SNIPEND-->
+
+Notice that the Worker will listen to the same Task Queue that the Workflow and Activity tasks are sent to. This is called "Task routing", and is a built-in mechanism for load balancing.
+
+<!--SNIPSTART money-transfer-project-template-java-shared-constants-->
+<!--SNIPEND-->
+
+Let's start the Worker. From within IntelliJ run the TransferMoneyWorker class, or from the project root, run the following command:
+
+```
+./gradlew startWorker
+```
+
+When the Worker starts it begins polling the Task Queue. The first Task it finds is the Workflow Task, which it executes. Executing the Workflow causes the Activity Tasks to be sent to the Task Queue as well. The Worker then grabs each of the Activity Tasks in their respective order from the Task Queue and executes each of the corresponding Activities.
+
+<img class="docs-image-centered" src={require('../static/img/docs/confetti.png').default} />
+
+**Congratulations**, you have run a Temporal Workflow application!
+
+## Failure simulation &nbsp;&nbsp;![](/img/docs/warning.png)
+
+By running the application as is, at minimum you got a taste of one of Temporal's amazing value propositions: visibility into the Workflow and the status of the Workers executing the code. Let's explore another key value proposition, maintaining the state of a Workflow, by simulating some failures. Make sure your Worker is stopped before proceeding.
+
+### Server crash
+
+In most modern application environments, to accommodate a host failure, leader election processes are applied to multiple application instances which are backed by databases that maintain the application state and so on. With Temporal, even if the Temporal server is down, the state of the Workflow is preserved. You can easily test this by following these steps (again, make sure your Worker is stopped so your Workflow doesn't finish):
```

Comment:
```suggestion
In most modern application environments, to accommodate a host failure, leader election processes are applied to multiple application instances which are backed by databases that maintain the application state and so on. With Temporal, even if the Temporal server is down, the state of the Workflow is preserved. You can easily test this by following these steps (again, make sure your Worker is stopped so your Workflow doesn't finish):
```

I'm not sure explaining leader election helps. On one hand it's definitely complex and we hide that. But also the sentence overwhelemed me.  

Maybe, "Unlike modern applications requirements which require complex leader election processes and external databases to handle failure...